directives/btn-group/btn-group.tpl.html

```diff
@@ -0,0 +1,3 @@
+<div class="wfm-btn-group" ng-class="{'material-depth-1':$ctrl.btnClass == 'wfm-btn-default'}">
+  <button ng-repeat="item in $ctrl.items" class="wfm-btn {{$ctrl.btnClass}}" ng-class="{'{{$ctrl.selectionClass}}': $ctrl.selected === item, 'first':$first, 'last':$last}" ng-click="$ctrl.output(item); $ctrl.selected = item">{{item}}</button>
```

Comment:
I just think `$ctrl.selected = item` is not necessary, or is it? I am thinking we should notify the change through `output` without changing `select` in our realm, let the parent component to change the value, and then the change will propagate to this component through the one-way bound `selected`

Reply:
i think it will be simpler to use if the behaviour is contained inside the component. That way there is no risk of anyone forgetting to set the selected item or getting it wrong.
Thats also why i didnt use the callback at first so that the component would just return the selected item and nothing else

Reply:
It would, but then it would violate the rule of one-way data flow. It should give away the responsibility of updating the value. Yes, developers would have to remember setting the value by themselves, but that would be explicit, which is nice. And we would not need to update `selected` in `onInit` as well.

Reply:
I dont understand, lets take it from the top.
The cases i saw was:
- A button should optionally be able to be preselected on load (like the day option in teams)
- If no such option is provided no item should be selected on load
- There should be a custom function run on click so we can do state changes and similar things
- It should work with both raised buttons and flat buttons

My reason for using the onInit is to detect if the developer has provided a preselected item. If not then nothing should be selected.
My reason for setting $ctrl.internalSelected = item on click is to update it so that the correct button is marked on click.  This internal one was never ment to be returned.

Reply:
My question for using `onInit` is since `selected` is provided from the outer component, we do not need to notify the outer in `onInit` again. If the outer should do anything, it should do without the inner's notifying. 

As for `$ctrl.internalSelected = item`, I just feel like not holding states inside the component (aka `internalSelected`) and making the component display states only. In other words, the correct button is marked not on click immediately but after the component notifies the outer one, and the outer one changes `selected` and the change 'flows back' to the component. But the current code does the work well for sure.

Reply:
Can we merge it and make modifications later or should we reject this pull request and start over?

Reply:
Yes! Let's merge it!